spec/support/controller_macros.rb

```diff
@@ -37,6 +37,19 @@ def answers_vote_request(answer, action='liked')
 
     def questions_vote_request(question, action='liked')
       post :vote, params: { id: question.id, vote: action }, as: :turbo_stream
-    end 
+    end
+
+    def comment_new_request(question, answer=nil)
+      return get :new_for_question, params: { question_id: question.id } if answer.nil?
+      get :new_for_answer, params: { question_id: question.id, answer_id: answer.id } 
```

Comment:
Please consider this:
if answer.nil?
  get :new_for_question, params: { question_id: question.id }
else
  get :new_for_answer, params: { question_id: question.id, answer_id: answer.id } 
end

(just as you did with post requests)